src/backend/marsha/core/tests/test_lti.py

```diff
@@ -121,6 +123,48 @@ def test_lti_passport_unknown(self, mock_verify):
             lti.verify()
         self.assertFalse(mock_verify.called)
 
+    @mock.patch.object(LTIOAuthServer, "verify_request", return_value=True)
+    def test_lti_passport_mismatched_referer(self, *_):
+        """Launch request with a referer that differs from the consumer_site should faild."""
+        ConsumerSiteLTIPassportFactory(
+            oauth_consumer_key="ABC123",
+            shared_secret="#Y5$",
+            consumer_site__domain="example.com",
+        )
+        data = {
+            "resource_link_id": "df7",
+            "context_id": "course-v1:ufr+mathematics+0001",
+            "roles": "Student",
+            "oauth_consumer_key": "ABC123",
+        }
+        request = self.factory.post(
+            "/", data, HTTP_REFERER="https://not-example.com/route"
+        )
+        lti = LTI(request)
+        with self.assertRaises(LTIException):
+            lti.verify()
```

Comment:
add message as commented below and check it to be sure it's the LTIException you expect?